src/System.Security.AccessControl/tests/Ace.Tests.cs

```diff
@@ -0,0 +1,482 @@
+            Assert.Throws<ArgumentNullException>("binaryForm", () => CompoundAce.CreateFromBinaryForm(null, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => CompoundAce.CreateFromBinaryForm(new byte[1], -1));
+            Assert.Throws<ArgumentException>("binaryForm", () => CompoundAce.CreateFromBinaryForm(new byte[ace.BinaryLength + 1], 2));
+            Assert.Throws<ArgumentException>("binaryForm", () => CompoundAce.CreateFromBinaryForm(new byte[ace.BinaryLength], 1));
+        }
+
+        [Fact]
+        public void CompoundAce_GetBinaryForm_Invalid()
+        {
+            CompoundAce ace = (CompoundAce)CompoundAce_CreateTestData(0, 1, 1, "S-1-5-11", 0)[0];
+            Assert.Throws<ArgumentNullException>("binaryForm", () => ace.GetBinaryForm(null, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => ace.GetBinaryForm(new byte[1], -1));
+            Assert.Throws<ArgumentOutOfRangeException>("binaryForm", () => ace.GetBinaryForm(new byte[ace.BinaryLength + 1], 2));
+            Assert.Throws<ArgumentOutOfRangeException>("binaryForm", () => ace.GetBinaryForm(new byte[ace.BinaryLength], 1));
+        }
+
+        [Theory]
+        [MemberData(nameof(CompoundAce_TestObjects))]
+        public void CompoundAce_GetBinaryForm(GenericAce testAce, byte[] expectedBinaryForm, int testOffset)
+        {
+            byte[] resultBinaryForm = new byte[testAce.BinaryLength + testOffset];
+            testAce.GetBinaryForm(resultBinaryForm, testOffset);
+            GenericAce_VerifyBinaryForms(expectedBinaryForm, resultBinaryForm, testOffset);
+        }
+
+        [Theory]
+        [MemberData(nameof(CompoundAce_TestObjects))]
+        public void CompoundAce_CreateFromBinaryForm(GenericAce expectedAce, byte[] testBinaryForm, int testOffset)
+        {
+            GenericAce resultAce = CompoundAce.CreateFromBinaryForm(testBinaryForm, testOffset);
+            GenericAce_VerifyAces(expectedAce, resultAce);
+        }
+    }
+
+    public abstract class QualifiedAce_Tests : KnownAce_Tests
+    {
+
+    }
+
+    public class CommonAce_Tests : QualifiedAce_Tests
+    {
+        private static object[] CommonAce_CreateTestData(int intFlags, int intQualifier, int accessMask, string stringsid, bool isCallback, int opaqueLength, int offset)
+        {
+            AceFlags flags = (AceFlags)intFlags;
+            AceQualifier qualifier = (AceQualifier)intQualifier;
+            SecurityIdentifier sid = new SecurityIdentifier(stringsid);
+            byte[] opaque = new byte[opaqueLength];
+
+            CommonAce ace = new CommonAce(flags, qualifier, accessMask, sid, isCallback, opaque);
+            Assert.Equal(flags, ace.AceFlags);
+            Assert.Equal(accessMask, ace.AccessMask);
+            Assert.Equal(sid, ace.SecurityIdentifier);
+            Assert.Equal(opaque, ace.GetOpaque());
+            Assert.Equal(qualifier, ace.AceQualifier);
+            Assert.Equal(isCallback, ace.IsCallback);
```

Comment:
I think I already wrote a ValidateGenericAce method for that purpose but decided against using it for some reason. I'll put it back in.